Resfeber/View Controllers/ItineraryViewController.swift

```diff
@@ -10,22 +10,89 @@ import UIKit
+        collectionView.register(DestinationCell.self, forCellWithReuseIdentifier: DestinationCell.reuseIdentifier)
+        collectionView.dataSource = self
+        collectionView.delegate = self
+        view.addSubview(collectionView)
+        collectionView.anchor(top: view.safeAreaLayoutGuide.topAnchor,
+                              left: view.leftAnchor,
+                              bottom: view.bottomAnchor,
+                              right: view.rightAnchor,
+                              paddingTop: 12,
+                              paddingLeft: 20,
+                              paddingRight: 20)
+        
+        // Load Data
+        DestinationController.readDestinations()
+        collectionView.reloadData()
+    }
+}
+
+// MARK: - Collection View Layout
+
+extension ItineraryViewController: UICollectionViewDelegateFlowLayout {
+    func collectionView(_ collectionView: UICollectionView, layout _: UICollectionViewLayout, sizeForItemAt _: IndexPath) -> CGSize {
+        let width = collectionView.frame.size.width
+        let xInsets: CGFloat = 16
+        let cellWidth = width - (xInsets * 2)
+        let cellHeight: CGFloat = cellWidth * 0.65
+        return CGSize(width: cellWidth, height: cellHeight)
+    }
+
+    func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, minimumLineSpacingForSectionAt section: Int) -> CGFloat {
+        20
+    }
+}
 
-    /*
-    // MARK: - Navigation
+// MARK: - Collection View Data Source
 
-    // In a storyboard-based application, you will often want to do a little preparation before navigation
-    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        // Get the new view controller using segue.destination.
-        // Pass the selected object to the new view controller.
+extension ItineraryViewController: UICollectionViewDataSource {
+    func collectionView(_: UICollectionView, numberOfItemsInSection _: Int) -> Int {
+        DestinationData.itineraryDestinations.count
     }
-    */
 
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: DestinationCell.reuseIdentifier, for: indexPath) as! DestinationCell
+        guard indexPath.row < DestinationData.itineraryDestinations.count else { return DestinationCell() }
+
+        cell.destination = DestinationData.itineraryDestinations[indexPath.row]
+
+        return cell
+    }
+}
+
+// MARK: - Collection View Delegate
+
+extension ItineraryViewController: UICollectionViewDelegate {
+    func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
+        guard indexPath.row < DestinationData.itineraryDestinations.count else { return }
+        let destination = DestinationData.itineraryDestinations[indexPath.row]
+        print("DEBUG: Tapped destination: \(destination.name)..")
```

Comment:
And this one

Reply:
We will need this code in the future for presenting a detail view controller. In the meantime, I will replace the print statement with a `//TODO:` to help keep the codebase a little cleaner